### PR TITLE
fix(core): preserve query parameters in connection URLs for toolset binding

### PR DESCRIPTION
**Pull request by anubhav756**
## Overview

Fixes URL construction when initializing `ToolboxClient` with query parameters (e.g. `http://localhost:5000?num_rows=2` for parameter binding). Previously, naive string template concatenation (`${baseUrl}/mcp/` and `${this._mcpBaseUrl}${toolsetName || ''}`) corrupted base URLs containing query strings (yielding invalid URLs like `http://localhost:5000?num_rows=2/mcp/`).

> [!NOTE]
> This PR is a mirror of Python's https://github.com/googleapis/mcp-toolbox-sdk-python/pull/717.

## Changes

   - Refactored `constructor` using the `URL` API to ensure the `/mcp/` path is inserted before query parameters (e.g. `http://api.com?proj=xyz` ➔ `http://api.com/mcp/?proj=xyz`).
   - Added `protected appendToolsetPath(toolsetName?: string): string` helper method to join toolset path segments while preserving query string parameters.
   - Replaced string template URL concatenation in `toolsList()` with `this.appendToolsetPath(toolsetName)`.
   - Added `urlTestCases` parameterized unit tests in `test.base.ts` to verify base URL construction across URL edge cases.
   - Added `TestRunToolURLBinding` integration test case in `test.e2e.ts` verifying native server-side parameter injection via URL query binding.

### CODE REVIEW

File: packages/toolbox-core/src/toolbox_core/mcp/transportBase.ts [hunk 1]
```diff
@@ -61,7 +61,16 @@ export abstract class McpHttpTransportBase implements ITransport {
     clientName?: string,
     clientVersion?: string,
   ) {
-    this._mcpBaseUrl = `${baseUrl}/mcp/`;
+    const parsed = new URL(baseUrl);
+    let path = parsed.pathname;
+    if (path.endsWith('/mcp')) {
+      path += '/';
+    } else if (!path.endsWith('/mcp/') && !path.includes('/mcp/')) {
```
Comment (twishabansal) [inline packages/toolbox-core/src/toolbox_core/mcp/transportBase.ts]:
```suggestion
    } else if !path.includes('/mcp/')) {
```

**Review by twishabansal — APPROVED** [review]

**Review by twishabansal — APPROVED** [review]

File: packages/toolbox-core/src/toolbox_core/mcp/transportBase.ts (hunk 1 above)
Comment (anubhav756) [reply packages/toolbox-core/src/toolbox_core/mcp/transportBase.ts]: Thanks for catching this! Resolved and in Python SDK.
